lib/Graph/Node.cpp

```diff
@@ -14,6 +14,8 @@
  * limitations under the License.
  */
 
+#include <atomic>
```

Comment:
I don't think that a global variable is a good idea. I think that it should be a member of Module/Context.

Reply:
agreed, this is still WIP